### ISSUE TITLE
Feature request: Smart "ESC" key

### ISSUE THREAD

**Issue by scottjl** [UltimateHackingKeyboard/firmware#116]
Feature request I have with another keyboard (V60) of mine.

The ESC key is "smart". Pressed on it's own it is ESC, shift-keypress is ~, shift to mod-layer to type `
Very useful for people who spend most of their time in *nix/vim.

I can sort-of emulate this by remapping the physical key to ESC, and mod-key to ~ and Fn to ` but I can't map shift to a layer.

Thanks!

**Comment by mondalaci** [comment]
@scottjl I'm not sure what you're looking for. Is the following right?

* Tap the Esc key -> Esc
* Hold Shift, then press Esc -> ~
* Hold Mod, then press Esc -> `

**Comment by scottjl** [comment]
Yes, that would be it. I use ESC most, ~ less frequently and ` even less frequently. My older V60 keyboard has this type of binding for the ESC key and I found it incredibly useful. I'd replicate it on the UHK if I could but I can't use shift as a modifier in Agent, unless I'm missing it somewhere? I understand this is a feature request and don't expect it to be anywhere near the top of your priority list. Thanks!

**Comment by mondalaci** [comment]
Thanks for the explanation! We plan to allow this via https://github.com/UltimateHackingKeyboard/agent/issues/617, so I'm closing this duplicate issue.

**Comment by lpetrus** [comment]
@scottjl  I do not know whether you still want this feature.  It still can not be done using the official firmware however I was able to get it working using the [UHK firmware from Karel Tucek](/github.com/kareltucek/firmware).

First I created a macro named Escape Handler:

![image](https://user-images.githubusercontent.com/2259252/85409711-7015f400-b566-11ea-8b14-c179a79c83e5.png)

Then I assigned the macro to my main layout:

![image](https://user-images.githubusercontent.com/2259252/85409659-5d032400-b566-11ea-9b4d-0f3ba93912b3.png)

And for the mod layer I mapped the tilde key instead of the esc key:

![image](https://user-images.githubusercontent.com/2259252/85410088-efa3c300-b566-11ea-9108-a92713907fad.png)

Edit: Updated the screenshots to display a fixed version of the macro.  The first version of the macro did not work correctly - after pressing the escape key it displayed ERR.
